pkg/controller/account/account_controller.go

```diff
@@ -254,134 +268,261 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 
-	// Fetch the Account instance
-	currentAcctInstance := &awsv1alpha1.Account{}
-	err = r.Client.Get(context.TODO(), request.NamespacedName, currentAcctInstance)
-	if err != nil {
-		if k8serr.IsNotFound(err) {
-			return reconcile.Result{}, nil
+	// If the account is BYOC, needs some different set up
+	if currentAcctInstance.Spec.BYOC {
+
+		reqLogger.Info("BYOC account")
+
+		//Set awsAccountID
+		awsAccountID = currentAcctInstance.Spec.AwsAccountID
+
+		// Get associated AccountClaim
+		accountClaim := &awsv1alpha1.AccountClaim{}
+
+		err := r.Client.Get(context.TODO(),
+			types.NamespacedName{Name: currentAcctInstance.Spec.ClaimLink, Namespace: currentAcctInstance.Spec.ClaimLinkNamespace},
+			accountClaim)
+		if err != nil {
+			if k8serr.IsNotFound(err) {
+				return reconcile.Result{}, nil
```

Comment:
If the claim is not found we just return and do nothing?

Reply:
Yes because then we cannot get the secret holding the BYOC IAM credentials.